### PR TITLE
Add multiprocessing to tests

### LINKED ISSUE #611: multiprocess pytest
implement multiprocessing in our tests

### PR DESCRIPTION
**Pull request by ksunden**
Closes #611

### CODE REVIEW

**Comment by ksunden** [comment]
Shh, pep8speaks, the `black` is coming for your complaints soon enough

**Comment by ksunden** [comment]
It doesn't save as much time as I had hoped, but it does save some.

Installing pytest-mp on Windows did not seem to work, and caused the tests to hang without running...
Currently added to `.travis.yml`, can add to the `contributing.rst` if desired.

**Comment by ksunden** [comment]
Also, please make sure to squash this, this is entirely too many very small commits (as test focused PRs tend to be, by their nature)

File: setup.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ def read(fname):
     tests_require=[
         "pytest",
         "pytest-cov",
+        # "pytest-mp",
```
Comment (untzag) [inline setup.py]: can we delete this, rather than just commenting it out?

**Pull request merged**